x-pack/legacy/plugins/infra/public/hooks/use_track_visit.tsx

```diff
@@ -0,0 +1,30 @@
+/*
```

Comment:
We should move this file to the shared observability plugin once we figure out the best way to share code from there when it requires another Kibana core plugin like this one does. Passing it in may work for now.

Reply:
Note: stack services seemed interested in having this exported from the UI Metric plugin so we'll leave it here for now and look to move it there soon.